### PR TITLE
OSX compilation fixes

### PR DESCRIPTION
**Pull request by seriema**
Thanks to the work of [Herdinger's fork](https://github.com/Herdinger/EmulationStation) I was able to get your fork to compile on OSX. [I asked them](https://github.com/Herdinger/EmulationStation/issues/14) if it was ok to contribute their OSX build fixes and they said yes.

Let me know if you prefer that I squash these commits. I tried to keep them as-is so history is kept and hopefully make future contributions between the two forks easier.

Alternatively I have another fork of RetroPie's EmulationStation where I rebased all of Herdingers work onto it: https://github.com/seriema/EmulationStation/commits/herdinger-rebase

(I used JetBrains CLion IDE to run and test the build)

### CODE REVIEW

File: es-core/src/animations/Animation.h [hunk 1]
```diff
@@ -5,6 +5,7 @@
 class Animation
 {
 public:
+    virtual ~Animation() {}
```
Comment (joolswills) [inline es-core/src/animations/Animation.h]: wrong indent (space vs tabs)

File: CMakeLists.txt [hunk 2]
```diff
@@ -68,15 +68,7 @@ if(MSVC)
...
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /MP") #multi-processor compilation
-endif()
-
-if(CMAKE_COMPILER_IS_GNUCXX)
-    #check for G++ 4.7+
-    execute_process(COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE G++_VERSION)
-    if (G++_VERSION VERSION_LESS 4.7)
-            message(SEND_ERROR "You need at least G++ 4.7 to compile EmulationStation!")
-    endif()
-
```
Comment (joolswills) [inline CMakeLists.txt]: Is gcc > 4.7 no longer needed with your changes ?

File: es-core/src/AudioManager.cpp [hunk 3]
```diff
@@ -3,7 +3,7 @@
 #include <SDL.h>
 #include "Log.h"
 
-std::vector<std::shared_ptr<Sound>> AudioManager::sSoundVector;
+std::vector<std::shared_ptr<Sound> > AudioManager::sSoundVector;
```
Comment (joolswills) [inline es-core/src/AudioManager.cpp]: is this whitespace change required ?

**Comment by joolswills** [comment]
Sorry for the delay in getting back to you. Not sure we need these fixes since the RetroPie ES has some RetroPie specific customisations and RetroPie doesn't work on OSX currently (It probably could with some changes though). Although I see that the GL fix is probably wanted.. I made a couple of comments anyway. Cheers.

**Comment by seriema** [comment]
No problem. I had grabbed the original commits as-is to make any future rebasing easier, but had to edit one of them to fix some merge conflicts that expected changes from other commits I didn't want to include. So they might include things like extra spaces or what not, but I'll clean that up (tonight, hopefully).

RetroPie might not work on OSX, but ES can and supporting OSX might open up for more contributors (me for example). :) You can also see [the response on Herdinger's fork about me grabbing their commits for you](https://github.com/Herdinger/EmulationStation/issues/14#issuecomment-205047387). They've pulled in all your changes and I guess it still works, so it'd be nice if we get your fork of ES working on OSX.

For maintaining tabs/spaces and similar simple stuff between editors and contributors you could consider adding an [EditorConfig](http://editorconfig.org/) config file. I can create a PR for one if you'd like?

**Comment by joolswills** [comment]
I'm surprised they pulled in all of our changes - some are specifically for retropie.

File: es-core/src/AudioManager.cpp (hunk 3 above)
Comment (seriema) [inline es-core/src/AudioManager.cpp]: I guess it depends on the C++ version you're aiming for.
https://stackoverflow.com/questions/7087033/for-nested-templates-when-did-become-standard-c-instead-of

File: CMakeLists.txt (hunk 2 above)
Comment (seriema) [inline CMakeLists.txt]: Not my changes really so I can't say, but it could be related to the `> >` you commented on.

**Comment by seriema** [comment]
I've updated locally based on your comments, but it also makes me wonder about compiler support. I'd highly recommend setting up Travis and AppVeyor to make sure every commit and PR builds on your target platforms.

**Comment by seriema** [comment]
I pushed the space fix for `Animation.h` at least.

**Comment by joolswills** [comment]
This fork is maintained just for RetroPie as mentioned - so that would require someone willing to put some more time into the development of it. I don't think at this stage that is needed, but certainly if more work was being put in, it would be good to have that.

**Comment by joolswills** [comment]
Will test the latest changes when I next have a free moment and merge. thanks.

**Comment by seriema** [comment]
Setting up Travis and Appveyor is really easy, at least I think so. I could create the config files and then all you'd have to do is sign into the sites to copy paste the token for your account. Let me know if you're up for it.

**Comment by joolswills** [comment]
Does anyone have a problem with this - it's rather old and has been forgotten. Assuming it's still complete as code has changed a lot?

**Comment by joolswills** [comment]
The c compiler check may no longer be needed even.

**Comment by joolswills** [comment]
Ignore me. Seems GitHub has a mobile issue as it said this had no conflicts (cached from previous PR I checked maybe). I was surprised it didn't. I'm closing as it's long out of date

**Pull request closed without merging**